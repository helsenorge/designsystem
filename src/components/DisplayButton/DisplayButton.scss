// @import '../../styling/old-variables';

// .display-button {
//   position: relative;
//   display: inline-flex;
//   border: none;
//   background-color: transparent;
//   text-overflow: ellipsis;
//   color: white;
//   padding: 0 3.25rem;
//   min-height: 3.13rem;
//   font-size: 1.125rem;
//   align-items: center;
//   cursor: pointer;
//   appearance: none;
//   white-space: nowrap;
//   //FIX for IE11 to vertical-align elements
//   @media screen and (-ms-high-contrast: active), (-ms-high-contrast: none) {
//     height: 50px;
//   }
//   @media screen and (min-width: $one-to-two-col) {
//     .display-button {
//       &--fullwidth {
//         width: auto;
//       }
//     }
//   }
//   &:disabled {
//     pointer-events: none;
//     background-color: $color-base-4-2-5;
//   }
//   &--primary {
//     background-color: $color-primary-5;
//     &:hover,
//     &:focus {
//       background-color: $color-primary-5-5;
//       box-shadow: $outline-box-shadow-thin;
//       outline: none;
//     }
//     &:disabled {
//       box-shadow: none;
//       color: white;
//     }
//   }
//   &--secondary {
//     box-shadow: inset 0 0 0 3px $color-primary-5;
//     color: $color-primary-5;
//     &:hover,
//     &:focus {
//       box-shadow: $outline-box-shadow-thin, inset 0 0 0 3px $color-primary-5-5;
//       background-color: $color-base-4-5;
//       color: $color-primary-5-5;
//     }
//     &:disabled {
//       box-shadow: inset 0 0 0 3px $color-base-3-5;
//       color: $color-base-3-5;
//     }
//   }
//   &--tertiary {
//     color: $color-primary-5;
//     &:hover,
//     &:focus {
//       color: $color-primary-5-5;
//       background-color: $color-base-4-5;
//     }
//     &:disabled {
//       color: $color-base-3-5;
//     }
//   }
//   &--fullwidth {
//     width: 100%;
//   }
//   &__content {
//     flex-grow: 1;
//     text-align: center;
//     padding: 0rem;
//   }
//   &__icon-left {
//     padding-left: 0;
//     padding-right: 2rem;
//     // &.atom_displaybutton--hasrighticon {
//     //   padding-right: 0;
//     // }
//   }
//   &__icon-right {
//     padding-right: 0;
//     padding-left: 2rem;
//     // &.atom_displaybutton--haslefticon {
//     //   padding-left: 0;
//     // }
//   }
//   // .atom_displaybutton-icon-left,
//   // .atom_displaybutton-icon-right {
//   //   content: '';
//   //   height: 2rem;
//   //   width: 2rem;
//   //   padding: 0 2.25rem;
//   //   background-repeat: no-repeat;

//   //   background-size: contain;
//   // }

//   // .atom_displaybutton-icon-left {
//   //   background-position: 70%;
//   // }

//   // .atom_displaybutton-icon-right {
//   //   background-position: 30%;
//   // }
// }
