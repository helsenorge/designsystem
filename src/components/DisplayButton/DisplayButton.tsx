import React from 'react';
import cn from 'classnames';
import styles from './DisplayButton.scss';
import {ButtonVariants} from '../../constants';

interface DisplayButtonProps extends React.HTMLProps<HTMLButtonElement> {
  children: React.ReactNode;
  fullWidth?: boolean;
  leftIcon?: string;
  onClick?: () => void;
  rightIcon?: string;
  variant?: DisplayButtonVariants;
}

const DisplayButton = React.forwardRef<HTMLButtonElement, DisplayButtonProps>(
  ({children, variant = ButtonVariants.Primary, leftIcon, rightIcon, fullWidth, ...restProps}, ref): JSX.Element => {
    // function attachIconStyling(path?: string) {
    //   return path ? {background: `url(${path}) center no-repeat`} : {};
    // }

    const classes = cn(
      styles['display-button'],
      {[styles['display-button--primary']]: variant === ButtonVariants.Primary},
      {[styles['display-button--secondary']]: variant === ButtonVariants.Secondary},
      {[styles['display-button--tertiary']]: variant === ButtonVariants.Tertiary},
      {[styles['display-button--full-width']]: fullWidth},
    );

    return (
      <button ref={ref} className={classes} {...restProps}>
        {React.Children.map(children, child => {
          return <span className={styles['display-button__content']}>{child}</span>;
        })}
        {/* {leftIcon ? <span className={styles['display-button__icon-left']} style={attachIconStyling(leftIcon)} /> : null} */}
        {/* {React.Children.map(children, (child => {
        console.log('child', child);
      }))} */}
        {/* <span className={styles['display-button__content']}>{children}</span>
      {rightIcon ? (
        <span className={styles['display-button__icon-right']} style={attachIconStyling(rightIcon)} />
      ) : null} */}
      </button>
    );
  },
);

DisplayButton.defaultProps = {
  type: 'button',
};

type DisplayButtonVariants = (typeof ButtonVariants)[keyof typeof ButtonVariants];

export {DisplayButton};
