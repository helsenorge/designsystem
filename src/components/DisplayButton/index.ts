export {DisplayButton, DisplayButtonVariants} from './DisplayButton';
