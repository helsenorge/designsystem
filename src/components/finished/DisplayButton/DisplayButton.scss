@import '../../../styling/old-variables';

.display-button {
  position: relative;
  display: inline-flex;
  border: none;
  background-color: transparent;
  text-overflow: ellipsis;
  color: $color-base-0;
  padding: 0 3.25rem;
  min-height: 3.13rem;
  font-size: 1.125rem;
  text-align: center;
  align-items: center;
  cursor: pointer;
  appearance: none;
  white-space: nowrap;
  //FIX for IE11 to vertical-align elements
  @media screen and (-ms-high-contrast: active), (-ms-high-contrast: none) {
    height: 50px;
  }
  &:hover {
    box-shadow: $outline-box-shadow-thin;
  }
  &:focus {
    box-shadow: $outline-box-shadow-thin;
    outline: none;
  }
  &:disabled {
    pointer-events: none;
    background-color: $color-base-4-2-5;
  }
  &--haslefticon {
    padding-left: 0;
    padding-right: 2rem;
    &.atom_displaybutton--hasrighticon {
      padding-right: 0;
    }
  }
  &--hasrighticon {
    padding-right: 0;
    padding-left: 2rem;
    &.atom_displaybutton--haslefticon {
      padding-left: 0;
    }
  }

  &--fullwidth {
    width: 100%;
  }

  &--primary {
    background-color: $color-primary-5;

    &:hover,
    &:focus {
      background-color: $color-primary-5-5;
    }

    &:disabled,
    &.disabled {
      box-shadow: none;
      color: white;
    }
  }

  &--secondary {
    box-shadow: inset 0 0 0 3px $color-primary-5;
    color: $color-primary-5;

    &:hover,
    &:focus {
      box-shadow: $outline-box-shadow-thin, inset 0 0 0 3px $color-primary-5-5;
      color: $color-primary-5-5;
    }
    &:focus {
      background-color: $color-base-4-5;
    }

    &:disabled,
    &.disabled {
      box-shadow: inset 0 0 0 3px $color-base-3-5;
      color: $color-base-3-5;
    }
  }

  &--tertiary {
    color: $color-primary-5;

    &:hover,
    &:focus {
      color: $color-primary-5-5;
    }
    &:focus {
      background-color: $color-base-4-5;
    }

    &:disabled,
    &.disabled {
      color: $color-base-3-5;
    }
  }

  .atom_displaybutton-text {
    flex-grow: 1;
    text-align: center;
  }

  .atom_displaybutton-icon-left,
  .atom_displaybutton-icon-right {
    content: '';
    height: 2rem;
    width: 2rem;
    padding: 0 2.25rem;
    background-repeat: no-repeat;

    background-size: contain;
  }

  .atom_displaybutton-icon-left {
    background-position: 70%;
  }

  .atom_displaybutton-icon-right {
    background-position: 30%;
  }
}

@media screen and (min-width: $one-to-two-col) {
  .atom_displaybutton {
    &--fullwidth {
      width: auto;
    }

    &--primary {
    }

    &--secondary {
    }

    &--tertiary {
    }
  }
}
