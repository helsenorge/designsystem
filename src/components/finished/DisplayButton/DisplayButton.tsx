import React from 'react';
import cn from 'classnames';
import styles from './DisplayButton.scss';
import {ButtonVariants} from '../../../constants';

interface DisplayButtonProps extends React.HTMLProps<HTMLButtonElement> {
  children: React.ReactNode;
  fullWidth?: boolean;
  leftIcon?: string;
  onClick?: () => void;
  rightIcon?: string;
  variant?: DisplayButtonVariants;
}

function DisplayButton({
  children,
  variant = ButtonVariants.Primary,
  leftIcon,
  rightIcon,
  fullWidth,
  ...restProps
}: DisplayButtonProps): JSX.Element {
  function attachIconStyling(path?: string) {
    return path ? {background: `url(${path}) center no-repeat`} : {};
  }

  const classes = cn(
    styles['display-button'],
    {[styles['display-button--primary']]: variant === ButtonVariants.Primary},
    {[styles['display-button--secondary']]: variant === ButtonVariants.Secondary},
    {[styles['display-button--tertiary']]: variant === ButtonVariants.Tertiary},
    {[styles['display-button--full-width']]: fullWidth},
  );

  return (
    <button className={classes} {...restProps}>
      {leftIcon ? <span className={styles['display-button__icon-left']} style={attachIconStyling(leftIcon)} /> : null}
      <span className={styles['display-button__content']}>{children}</span>
      {rightIcon ? (
        <span className={styles['display-button__icon-right']} style={attachIconStyling(rightIcon)} />
      ) : null}
    </button>
  );
}

DisplayButton.defaultProps = {
  type: 'button',
};

export type DisplayButtonVariants = (typeof ButtonVariants)[keyof typeof ButtonVariants];
export default DisplayButton;
