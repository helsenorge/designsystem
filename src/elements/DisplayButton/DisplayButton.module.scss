@import '../../styling/colors';

.display-button {
  position: relative;
  display: inline-flex;
  border: none;
  border-radius: 0;
  background-color: $secondary-color;
  text-overflow: ellipsis;
  letter-spacing: normal;
  color: white;
  margin: 0;
  padding: 0 3.25rem;
  min-height: 3.13rem;
  font-size: 1.125rem;
  text-align: center;
  align-items: center;
  cursor: pointer;
  appearance: none;
  white-space: nowrap;
  &:focus,
  &:hover {
    background-color: $secondary-color-hover;
    box-shadow: 0 0 0 4px $outline;
  }
  &--secondary {
  }
  &--tertiary {
  }
}
