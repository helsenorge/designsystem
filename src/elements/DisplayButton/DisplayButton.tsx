import cn from 'classnames';
import React from 'react';
import styles from './DisplayButton.module.scss';

type DisplayButtonVariant = 'primary' | 'secondary' | 'tertiary';

interface DisplayButtonProps extends React.HTMLProps<HTMLButtonElement> {
  children: React.ReactNode;
  onClick?: () => void;
  variant?: DisplayButtonVariant;
}

function DisplayButton(props: DisplayButtonProps) {
  const classes = cn(styles['display-button']);
  return (
    <button className={classes} {...props}>
      {props.children}
    </button>
  );
}

DisplayButton.defaultProps = {
  type: 'button',
  variant: 'primary',
};

export {DisplayButton};
