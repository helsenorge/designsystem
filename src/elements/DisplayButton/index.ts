import {DisplayButton} from './DisplayButton';

export default DisplayButton;
