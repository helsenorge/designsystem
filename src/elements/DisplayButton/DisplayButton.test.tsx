import React from 'react';
import {render} from 'react-testing-library';
import DisplayButton from '.';

test('default variant renders correctly', () => {
  const {container} = render(<DisplayButton>Hello World</DisplayButton>);
  expect(container).toMatchSnapshot();
});
