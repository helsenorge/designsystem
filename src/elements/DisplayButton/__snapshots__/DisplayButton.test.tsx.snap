// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default variant renders correctly 1`] = `
<div>
  <button
    class="display-button"
    type="button"
    variant="primary"
  >
    Hello World
  </button>
</div>
`;
